### PR TITLE
improved logrus output switching performance

### PR DESCRIPTION
**Pull request by NyaaaWhatsUpDoc**
The previous method of looking for level string in the log bytes would perform a very simplistic search for the string `level=error`. So for a log message with `level=info msg="hello world"` it would index past the `level=info`, and keep looking along the rest all the way until the end. This new code makes the output splitting more customizable (and so now includes error, fatal and panic to stderr), makes it more testable, and improves performance by only checking along the log bytes until the first instance of `level=`.

### CODE REVIEW

File: internal/log/log_test.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package log_test
```
[the remark this answers is not in the document]
Comment (NyaaaWhatsUpDoc) [reply internal/log/log_test.go:19]: done :)

**Comment by NyaaaWhatsUpDoc** [comment]
@tsmethurst any idea why the test for this has suddenly failed simply by adding the license comment at the top of the file? Console logs don't provide any useful info.

**Comment by tsmethurst** [comment]
> @tsmethurst any idea why the test for this has suddenly failed simply by adding the license comment at the top of the file? Console logs don't provide any useful info.

A race condition in the test i think; kicking fixed it ;)

**Comment by igalic** [comment]
is the race-condition just in the test code, or also in the tested code?

**Pull request merged**